drivers/hmis/app/graphql/types/hmis_schema/enrollment.rb

```diff
@@ -267,6 +268,12 @@ def source_referral_posting
       load_ar_association(object, :source_postings).min_by(&:id)
     end
 
+    def source_ce_referral
+      # Skip checking permission here, because the current user can view the referral's target enrollment (this object),
+      # so we know they have at least can_view_summary? on the referral.
+      load_ar_association(object, :source_ce_referral)
+    end
```

Comment:
I think it would probably be worthwhile to make this policy check explicit; It shouldn't be a big cost because of the caching. That way if we change the logic of the can_view_summary method, we can track it back to this use-case, avoiding a scenario where can_view_summary became more restrictive but we still attempt to resolve it.

```suggestion
    def source_ce_referral
      source_referral = load_ar_association(object, :source_ce_referral)
      return unless source_referral # enrollment did not originate from a ce referral

      policy = current_user.policy_for(source_referral, policy_type: :ce_referral)
      source_referral if policy.can_view_summary?
    end
```